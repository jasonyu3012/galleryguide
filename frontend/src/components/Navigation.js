import 'bootstrap/dist/css/bootstrap.min.css';
import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import { Link } from 'react-router-dom';

const Navigation = () => {
  return (
    <Navbar bg="light" expand="lg">
      <Container>
        <Navbar.Brand as={Link} to="/">GalleryGuide</Navbar.Brand>
          <Nav className="me-auto">
            <Nav.Link as={Link} to="/">Home</Nav.Link>
            <Nav.Link as={Link} to="/about">About</Nav.Link>
            <Nav.Link as={Link} to="/artworks">Artworks</Nav.Link>
            <Nav.Link as={Link} to="/artists">Artists</Nav.Link>
            <Nav.Link as={Link} to="/galleries">Galleries</Nav.Link>
          </Nav>
          <input
            // ref={inputRef}
            type="text"
            placeholder="Site-wide search"
            id="message"
            name="message"
            // value={searchInput}
            // onChange={(e) => searchItems(e.target.value)}
           />

      </Container>
    </Navbar>
  );
}

export default Navigation;