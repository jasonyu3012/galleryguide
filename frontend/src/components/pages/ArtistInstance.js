// React imports
import { Link, useParams } from 'react-router-dom';
import React, { useState, useEffect } from 'react';
// Library imports
import axios from 'axios';
import Button from 'react-bootstrap/Button';
// Local imports
import { ARTISTS_NUM_IDS } from './Artists';
import './InstanceModels.css';
import { googleKey } from '../../API_keys/Keys.js';


const ArtistInstance = () => {
  const artistId = useParams().artistId;
  // Note on naming conventions: `data` is for the main information on the instance cards. `info` is
  // for the supplementary info (not the main point).
  const [artistData, setArtistData] = useState({});
  const [artworkId, setArtworkId] = useState(0);
  const [galleryId, setGalleryId] = useState(0);
  const [videoSrc, setVideoSrc] = useState("");
  const [mapSrc, setMapSrc] = useState("");
  const [artworkData, setArtworkData] = useState({});
  const [galleryData, setGalleryData] = useState({});
  // Run only once due to second arg
  useEffect(() => {
    console.log("page loaded")
    axios.get(`https://galleryguide.me/api/artists/${ artistId }`, {
    params : {
      gallery_ids: 1 , 
      artwork_ids : 1
      }
    })
    .then(response => {
      const artistData = response.data;
      setArtistData(artistData);
      console.log("response data: ", artistData);

      // Get google maps media src
      // Take the title, split by spaces, join by browser space code %20
      const mapQuery = artistData.name.replace(' ', "%20")
      setMapSrc(`https://www.google.com/maps/embed/v1/place?q=${ mapQuery }&key=${ googleKey.key }`);
      console.log("map src: ", mapSrc);

      // Get youtube media
      console.log("getting src for gallery ", artistData.name)
      axios.get(`https://www.googleapis.com/youtube/v3/search?part=snippet&maxResults=1&q=${ artistData.name }&type=video&key=${ googleKey.key }`)
      .then(response => {
        console.log("youtube video data: ", response.data);
        const fullSrc = `https://www.youtube.com/embed/${ response.data.items[0].id.videoId }`
        setVideoSrc(fullSrc)
        console.log("id in request: ", response.data.items[0].id.videoId)
        console.log("src in request: ", fullSrc);
      })
      .catch((error) => {
        console.log("axios error when fetching video: ", error);
        setVideoSrc(`https://www.youtube.com/embed/fhdZxedNFr0`);
      })

      // Get artwork and gallery IDs
      if (artistData.artwork_ids) {
        setArtworkId(artistData.artwork_ids[Math.floor(Math.random() * artistData.artwork_ids.length)])
          axios.get(`https://galleryguide.me/api/artworks/${ artistId }`, { 
          params: {
        }})
        .then(response => {
          const responseData = response.data
          setArtworkData({responseData})
        })
      }
      if (artistData.gallery_ids) {
        setGalleryId(artistData.gallery_ids[Math.floor(Math.random() * artistData.gallery_ids.length)])
      }
    })
    .catch((error) => {
      console.log("axios error: ", error);
    })
  }, [])

  // Check that we got a valid ID request
  if (artistId < 1 || artistId > ARTISTS_NUM_IDS) {
    return (
      <div>
        <h1>Oops...!</h1>
        {/* Tara's note: &#41; is the html code for a closing parenthesis, ) */}
        <p>How did you get here? We don't have an artist with identification #{ artistId }. Let us know if you think this is our error. :&#41;</p>
      </div>
    );
  }
 
  // Tara's note: there's a lot more potential data we can add in Phase 3 to make this look prettier!
  //needs to be changed to be relevant to artists
  return (
    <div>
      <div class="art-instance-image-description-wrapper">
        <img class="art-instance-img" src={ artistData.thumbnail } alt="Artwork visual representation"/>
        <div class="art-instance-description">
          <h1>{ artistData.name }</h1>
          {/* Death year is null if still alive */}
          <h3><em>from { artistData.birth_year } to { artistData.death_year ? artistData.death_year : "Present" }</em></h3>
          <p>Number of artworks made: { artistData.num_artworks }</p>
          <p>Number of associated galleries: { artistData.num_galleries }</p>
          <br/>
          <br/>
        </div>
      </div>
      <p>{ artistData.biography }</p>
      <br/>
      <br/>
      <p>This artist is stored as ID #{ artistData.id } in GalleryGuide.</p>
      {/* INSTANCE CONNECTIONS  */}
      <Link to={`/artworks/${ artworkId }`}>
        <Button>Explore Random Artwork: {artworkData.name}</Button>
      </Link>
      <p></p>
      <Link to={`/galleries/${ galleryId }`}>
        <Button>Explore Related Gallery</Button>
      </Link>

      {/* More media */}
      <div>
        <h2>Media for a more immersive experience</h2>
        <p>Note that these results are generated by a search. Results are not curated and GalleryGuide takes no ownership of generated results.</p>
        <iframe width="600" height="450" src={ videoSrc } style={{marginRight: '3em'}}></iframe>
        <iframe width="600" height="450" src={ mapSrc }></iframe>
      </div>

      {/* TODO #? Add connections to other instances */}

      <Link to={'/artists'}>
        <Button>&#60;</Button> Back to main artworks page
      </Link>
    </div>
  )
};

export default ArtistInstance;
