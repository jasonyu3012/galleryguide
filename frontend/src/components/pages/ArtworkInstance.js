// React imports
import { Link, useParams } from 'react-router-dom';
import React, { useState, useEffect } from 'react';
// Library imports
import axios from 'axios';
import Button from 'react-bootstrap/Button';
// Local imports
import { ARTWORKS_NUM_IDS } from './Artworks';
import './Artworks.css';


const ArtworkInstance = () => {
  const artworkId = useParams().artworkId;
  const [artworkData, setArtworkData] = useState({});
  const [artistInfo, setArtistInfo] = useState({});
  const [galleryInfo, setGalleryInfo] = useState({});

  // Run only once due to second arg
  useEffect(() => {
    console.log("page loaded")
    axios.get(`https://galleryguide.me/api/artworks/${ artworkId }`)
    .then(response => {
      const artworkData = response.data;
      console.log("artwork data: ", artworkData);
      setArtworkData(artworkData);
    })
    .catch((error) => {
      console.log("axios error: ", error);
    })

    // Get gallery and artist information
    if (artworkData) {
      axios.get(`https://galleryguide.me/api/galleries/${ artworkData.gallery_id }`)
      .then(response => {
        setGalleryInfo(response.data);
        console.log("gallery data: ", galleryInfo);
      })
      .catch((error) => {
        console.log("axios error while getting gallery info: ", error);
      })
      axios.get(`https://galleryguide.me/api/artists/${ artworkData.artist_id }`)
      .then(response => {
        setArtistInfo(response.data);
        console.log("artist data: ", artistInfo);
      })
      .catch((error) => {
        console.log("axios error while getting artist info: ", error);
      })
    }
  }, [])

  // Check that we got a valid ID request
  if (artworkId < 1 || artworkId > ARTWORKS_NUM_IDS) {
    return (
      <div>
        <h1>Oops...!</h1>
        {/* Tara's note: &#41; is the html code for a closing parenthesis, ) */}
        <p>How did you get here? We don't have an artwork with identification #{ artworkId }. Let us know if you think this is our error. :&#41;</p>
      </div>
    );
  }
 
  // Tara's note: there's a lot more potential data we can add in Phase 3 to make this look prettier!
  return (
    <div>
      <div class="art-instance-image-description-wrapper">
        <img class="art-instance-img" src={ artworkData.image } alt="Artwork visual representation"/>
        <div class="art-instance-description">
          <h1>{ artworkData.title }</h1>
          <h3>{ artistInfo.name }, { artworkData.date }</h3>
          <em>{ artworkData.medium }</em>
          <br/>
          <br/>
          <p>{galleryInfo ? `Artwork hosted at ${galleryInfo.name} in ${galleryInfo.region}` : null}</p>
          <p>{ artworkData.image_rights }</p>
        </div>
      </div>

      {/* TODO #? Add connections to other instances */}

      <Link to={'/artworks'}>
        <Button>&#60;</Button> Back to main artworks page
      </Link>
    </div>
  )
};

export default ArtworkInstance;
