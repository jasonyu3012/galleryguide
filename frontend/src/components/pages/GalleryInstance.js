// React imports
import { Link, useParams } from 'react-router-dom';
import React, { useState, useEffect } from 'react';
// Library imports
import axios from 'axios';
import Button from 'react-bootstrap/Button';
// Local imports
import { GALLERIES_NUM_IDS } from './GalleriesDynamic';
import './Artworks.css';


const GalleryInstance = () => {
  const galleryId = useParams().galleryId;
  const [artworkData, setArtworkData] = useState({});
  const [artistInfo, setArtistInfo] = useState({});
  const [galleryInfo, setGalleryInfo] = useState({});

  // Run only once due to second arg
  useEffect(() => {
    console.log("page loaded")
    axios.get(`https://galleryguide.me/api/galleries/${ galleryId }`)
    .then(response => {
      const galleryInfo = response.data;
      console.log("gallery data: ", galleryInfo);
      setGalleryInfo(galleryInfo);
    })
    .catch((error) => {
      console.log("axios error: ", error);
    })

    // Get artwork and artist information
    if (artworkData) {
      axios.get(`https://galleryguide.me/api/artworks/${ artworkData.artwork_id }`)
      .then(response => {
        setArtworkData(response.data);
        console.log("artwork data: ", artworkData);
      })
      .catch((error) => {
        console.log("axios error while getting artwork info: ", error);
      })
      axios.get(`https://galleryguide.me/api/artists/${ artworkData.artist_id }`)
      .then(response => {
        setArtistInfo(response.data);
        console.log("artist data: ", artistInfo);
      })
      .catch((error) => {
        console.log("axios error while getting artist info: ", error);
      })
    }
  }, [])

  // Check that we got a valid ID request
  if (galleryId < 1 || galleryId > GALLERIES_NUM_IDS) {
    return (
      <div>
        <h1>Oops...!</h1>
        {/* Tara's note: &#41; is the html code for a closing parenthesis, ) */}
        <p>How did you get here? We don't have an artwork with identification #{ artworkId }. Let us know if you think this is our error. :&#41;</p>
      </div>
    );
  }
 
  // Tara's note: there's a lot more potential data we can add in Phase 3 to make this look prettier!
  //change to be relevant to galleries
  return (
    <div>
      <div class="art-instance-image-description-wrapper">
        <img class="art-instance-img" src={ artworkData.image } alt="Artwork visual representation"/>
        <div class="art-instance-description">
          <h1>{ artworkData.title }</h1>
          <h3>{ artistInfo.name }, { artworkData.date }</h3>
          <em>{ artworkData.medium }</em>
          <br/>
          <br/>
          <p>{galleryInfo ? `Artwork hosted at ${galleryInfo.name} in ${galleryInfo.region}` : null}</p>
          <p>{ artworkData.image_rights }</p>
        </div>
      </div>

      {/* TODO #? Add connections to other instances */}

      <Link to={'/galleries'}>
        <Button>&#60;</Button> Back to main galleries page
      </Link>
    </div>
  )
};

export default GalleryInstance;
