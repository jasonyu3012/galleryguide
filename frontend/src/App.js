import React from 'react';
import {
  BrowserRouter,
  Route,
  Routes
} from 'react-router-dom';

import './App.css';
// Import pages
import About from './components/pages/About';
import Home from './components/pages/Home';
import Navigation from './components/Navigation';

import Artworks from './components/pages/Artworks';
import ArtworkInstance from './components/pages/ArtworkInstance';

import Artists from './components/pages/Artists';
import Galleries from './components/pages/Galleries';

import ExampleArtistInstance from './components/pages/ArtistInstances/ExampleArtistInstance';
import VincentvanGogh from './components/pages/ArtistInstances/VincentvanGogh';
import MaryCassatt from './components/pages/ArtistInstances/MaryCassatt';
import GrantWood from './components/pages/ArtistInstances/GrantWood';

import ExampleGalleryInstance from './components/pages/GalleryInstances/ExampleGalleryInstance';
import METMuseum from './components/pages/GalleryInstances/METMuseum';
import ArtInstituteofChicago from './components/pages/GalleryInstances/ArtInstituteofChicago';
import NationalGalleryofArt from './components/pages/GalleryInstances/NationalGalleryofArt';

function App() {
  return (
    <div className="app-wrapper">
      <BrowserRouter>
        {/* placement for the global navbar below (items to show up on every page) */}
        <Navigation />
        {/* placement for all of the internal pages below */}
        <Routes>
          <Route path="/" element={ <Home/> }/>
          <Route path="/about" element={ <About/> }/>
          <Route path="/artworks" element={ <Artworks/> }/>
          <Route path="/artworks/:artworkId" element={ <ArtworkInstance/> }/>
          <Route path="/artists" element={<Artists />} />
          <Route path="/galleries" element={<Galleries />} />

          <Route path="/artists/ExampleArtistInstance" element={<ExampleArtistInstance />} />
          <Route path="/artists/VincentvanGogh" element={<VincentvanGogh />} />
          <Route path="/artists/MaryCassatt" element={<MaryCassatt />} />
          <Route path="/artists/GrantWood" element={<GrantWood />} />

          <Route path="/galleries/ExampleGalleryInstance" element={<ExampleGalleryInstance />} />
          <Route path="/galleries/METMuseum" element={<METMuseum />} />
          <Route path="/galleries/NationalGalleryofArt" element={<NationalGalleryofArt />} />
          <Route path="/galleries/ArtInstituteofChicago" element={<ArtInstituteofChicago />} />

        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
