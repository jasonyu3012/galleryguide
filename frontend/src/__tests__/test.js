import renderer from "react-test-renderer";

import ArtworkInstance from "../components/Pages/ArtworkInstance";
import GalleryInstance from "../components/Pages/GalleryInstance";
import ArtistInstance from "../components/Pages/ArtistInstance";

import Navigation from "../components/Navigation";

import App from "../App";
import {
  BrowserRouter,
  Route,
  Routes,
} from 'react-router-dom';



it("displays the artwork correctly", () => {
  const artworkSample = {
    id: 1,
    title: "Title",
    date: "2022",
    medium: "Oil on canvas",
    iconicity: "High",
    image: "https://example.com/image.jpg",
    image_rights: "All rights reserved",
    artist_id: 1,
    gallery_id: 1,
  };

  const component = renderer.create(
    <BrowserRouter>
      <ArtworkInstance artworkData={artworkSample} />
    </BrowserRouter>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});



it("Navbar initializes correctly", () => {
  const component = renderer.create(
    <BrowserRouter>
      <Navigation />
    </BrowserRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

