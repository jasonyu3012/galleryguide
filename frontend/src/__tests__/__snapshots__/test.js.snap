// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar initializes correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      GalleryGuide
    </a>
    <div
      className="me-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rr-ui-event-key={null}
        disabled={false}
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="nav-link"
        data-rr-ui-event-key={null}
        disabled={false}
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
      <a
        className="nav-link"
        data-rr-ui-event-key={null}
        disabled={false}
        href="/artworks"
        onClick={[Function]}
      >
        Artworks
      </a>
      <a
        className="nav-link"
        data-rr-ui-event-key={null}
        disabled={false}
        href="/artists"
        onClick={[Function]}
      >
        Artists
      </a>
      <a
        className="nav-link"
        data-rr-ui-event-key={null}
        disabled={false}
        href="/galleries"
        onClick={[Function]}
      >
        Galleries
      </a>
    </div>
    <div>
      <input
        id="g-query"
        name="global-search-query"
        placeholder="Site-wide search"
        type="text"
      />
      <button
        style={
          Object {
            "marginLeft": "1em",
          }
        }
        type="submit"
      >
        Search
      </button>
    </div>
  </div>
</nav>
`;

exports[`displays the artwork correctly 1`] = `
<div>
  <div
    class="art-instance-image-description-wrapper"
  >
    <img
      alt="Artwork visual representation"
      class="art-instance-img"
    />
    <div
      class="art-instance-description"
    >
      <h1 />
      <h3>
        , 
      </h3>
      <em />
      <em>
        Iconicity of the image: 
      </em>
      <br />
      <br />
      <p />
      <br />
      <p>
        This artwork is stored as ID #
         in GalleryGuide.
      </p>
      <p />
      <a
        href="/artists/undefined"
        onClick={[Function]}
      >
        <button
          className="btn btn-primary"
          disabled={false}
          type="button"
        >
          Explore Artist
        </button>
      </a>
      <p />
      <a
        href="/galleries/undefined"
        onClick={[Function]}
      >
        <button
          className="btn btn-primary"
          disabled={false}
          type="button"
        >
          Explore Gallery
        </button>
      </a>
    </div>
  </div>
  <div>
    <h2>
      Media for a more immersive experience
    </h2>
    <p>
      Note that these results are generated by a search. Results are not curated and GalleryGuide takes no ownership of generated results.
    </p>
    <iframe
      height="450"
      src=""
      style={
        Object {
          "marginRight": "3em",
        }
      }
      width="600"
    />
    <iframe
      height="450"
      src=""
      width="600"
    />
  </div>
  <a
    href="/artworks"
    onClick={[Function]}
  >
    <button
      className="btn btn-primary"
      disabled={false}
      type="button"
    >
      &lt;
    </button>
     Back to main artworks page
  </a>
</div>
`;
